industrydive/sourcedive
# TECH-7631 admin.py clean-up and make more DRY try 2

[TECH-7631 cleanup admin.py and make it more DRY](https://industrydive.atlassian.net/browse/TECH-7631) -- I also fixed the bug where fieldsets were not appearing on the add page, and changed the privacy field to readonly in many cases. 

### Non-technical Context
Removed a workaround where you could change a semiprivate person to public and therefore expose their details. Cleaned up add page.

### Technical Context
There was a lot of duplicated code, and manipulating the readonly fields in particular causes problems. I pulled things out into the `get_fields`, `get_fieldsets` and `get_readonly_fields` hooks that django wants you to use.

### Manual Testing Instructions

Have 2 normal users. Log in as 1 of them and create 3 sources, one of each privacy level. The add page should be organized. Now view each source, and edit as well. You should see all the data, and everything except the default readonly fields should be editable.

Log in as the 2nd person. You should only see 2 of the suorces. View both. The appropriate fields should be hidden. Edit both. You should not be able to change the hidden fields or the privacy level of the semi private source.  

Create a semiprivate interaction as user1. You should be able to edit it. As user2, view it (you should not see the notes) and edit it (you should not be able to edit the privacy level OR the notes themselves).


### QA Checklist

#### Developer

- <s>Devise edge cases to test the bug fix or feature being merged</s> DECIDED NOT TO IN CONSULTATION WITH @greglinch 
    - [N/A] Technical documentation


#### Reviewer

    - If there are few or minor issues, add inline GitHub comments
    - If there are many issues or it seems like the developer did not follow the style guide, leave a comment asking them to do so
    - [na] Technical documentation
